Detectors/ITSMFT/ITS/tracking/src/VertexerTraits.cxx

```diff
@@ -66,7 +68,9 @@ void trackleterKernelSerial(
         // loop on clusters next layer
         for (int iNextLayerClusterIndex{ firstRowClusterIndex }; iNextLayerClusterIndex < maxRowClusterIndex && iNextLayerClusterIndex < (int)clustersNextLayer.size(); ++iNextLayerClusterIndex) {
           const Cluster& nextCluster{ clustersNextLayer[iNextLayerClusterIndex] };
-          const char testMC{ !isMc || (nextLayerMClabels[iNextLayerClusterIndex] == currentLayerMClabels[iCurrentLayerClusterIndex] && nextLayerMClabels[iNextLayerClusterIndex] != -1) };
+          const auto& lblNext = evt->getClusterLabels(layerIndex, nextCluster.clusterId);
```

Comment:
Without checking in detail, I would where possible try to keep the reconstruction code and the MC code separate. Don't know if this is possible here, but e.g. for the TPC I first create the tracks, then if MC is present I loop over the tracks and assign the labels. In addition I have some debug code inside the tracking that might access the labels, but that is usually disabled at compile time.

Reply:
Ok, I see what you mean.
For the moment the upstream code is giving wrong results as a consequence of the changes to labels.
I'd rather merge this in order to make code be correct.
Later I'll factorise the MC part out from the reco code, maybe changing it in a way it will work also with the GPU version.

Cheers